hbase-http/src/main/java/org/apache/hadoop/hbase/util/JSONBean.java

```diff
@@ -216,7 +239,12 @@ private static int write(JsonWriter writer, MBeanServer mBeanServer, ObjectName
       } else {
         MBeanAttributeInfo[] attrs = minfo.getAttributes();
         for (int i = 0; i < attrs.length; i++) {
-          writeAttribute(writer, mBeanServer, oname, description, attrs[i]);
+          if (matchingPattern != null) {
```

Comment:
Infact u can see we dont need such if else, end of the day we call same writeAttribute which is having Pattern[] arg and we have null check there anyways.